### ISSUE TITLE
[8.8.0] Repeating "non-compliant" license error in master logs after upgrade.

### ISSUE THREAD

**Issue by ceeeekay** [elastic/elasticsearch#96365]
### Elasticsearch Version

8.8.0

### Installed Plugins

n/a

### Java Version

_bundled_

### OS Version

Ubuntu 22.04.2 LTS

### Problem Description

After upgrading from 8.7.1 to 8.8.0, I'm seeing the following error repeating in the active master logs every 10 seconds:

```
[2023-05-26T16:47:27,929][WARN ][o.e.x.a.EnterpriseSearchUsageTransportAction] [redacted-hostname] Failed to get search application count to include in Enterprise Search usage
java.util.concurrent.ExecutionException: org.elasticsearch.ElasticsearchSecurityException: Current license is non-compliant for search application and behavioral analytics. Current license is active basic license. Search Applications and behavioral analytics require an active trial, platinum or enterprise license.
```
Along with a lot of stack trace.

This is occurring on multiple clusters that: 
* have a Basic license
* have never had a trial license activated
* don't make any use of Enterprise Search
* have been upgraded from 8.7.1 to 8.8.0
* have been around much longer than 8.7.1

I'm unable to reproduce this on a fresh 8.8.0 cluster, nor a fresh 8.7.1 cluster upgraded to 8.8.0. I could go back further with fresh installs but I really don't know how far that would need to be, or if it would even tell me anything.

It doesn't seem to be harming anything, however it's making a huge mess of my logs. Is there something I can do in the mean time to stop whatever is making the call that's running foul of the license check?

### Steps to Reproduce

I'm not sure how to test to reproduce this.

Existing clusters now on 8.8.0 are exhibiting this behaviour but I can no longer downgrade them to test anything.

Fresh 8.7.1 installations don't exhibit the issue when upgrading 8.7.1 -> 8.8.0, nor do fresh 8.8.0 installations.

**Comment by voran** [comment]
I had the same issue, and setting `xpack.monitoring.collection.enabled` to `false` solved this issue. However, keep in mind this disabled self-monitorinng.

**Comment by ceeeekay** [comment]
Yep that definitely stops the errors, but I use self-monitoring pretty extensively so it's not a solution unfortunately.

I don't understand why it would suddenly start giving errors related to Enterprise Search either - seems like a bug.

**Comment by markwalkom** [comment]
Also mentioned here https://discuss.elastic.co/t/elastic-8-8-0-current-license-is-non-compliant-for-search-application-and-behavioral-analytics/334712

**Comment by mashaikx13** [comment]
@voran I believe it won't affect the metricbeat stack monitoring right?

**Comment by mashaikx13** [comment]
I have tried your workaround but i am still receiving errors
```
PUT _cluster/settings
{
  "persistent": {
    "xpack.monitoring.collection.enabled": true
  }
}
```

**Comment by carlyrichmond** [comment]
This discuss question also looks to be caused by the same issue:

https://discuss.elastic.co/t/eck-continuous-log-spamming/334569/2

**Comment by voran** [comment]
> I have tried your workaround but i am still receiving errors
> 
> ```
> PUT _cluster/settings
> {
>   "persistent": {
>     "xpack.monitoring.collection.enabled": false
>   }
> }
> ```


@mashaikx13 I tried setting it permanently via the configuration file, not via the _cluster/settings endpoint.

**Comment by sbocquet** [comment]
Hi,

Same problem here after upgrading 8.7.1 to 8.8.0 this morning.
Plus, I'm unable to connect to Kibana with Edge but work with Firefox !...

**Comment by ceeeekay** [comment]
This seems to be fixed in 8.8.1. Two of my clusters with self-monitoring that have been upgraded to 8.8.1 are no longer logging these warnings.

I don't see anything specific in the release notes, but something seems to have solved it.

**Comment by C-monC** [comment]
I have upgraded to 8.8.1 and still get the issue.

**Comment by ceeeekay** [comment]
I'm not sure what the cause or solution was - an upgrade to 8.8.1 resolved it for me. I'll re-open this for you so maybe it gets some attention.

**Comment by carlosdelest** [comment]
Pinging @liranabn for prioritization and inclusion in backlog

**Comment by kderusso** [comment]
@liranabn - Looks like this might still be an issue we want to prioritize.

**Comment by C-monC** [comment]
Finally solved for me by upgrading to 8.9.0.

**Comment by dnhatn** [comment]
The excessive logging was removed in https://github.com/elastic/elasticsearch/pull/98071.
